spec/Order.md

```diff
@@ -0,0 +1,58 @@
+| expiration      | U64             | Time at which the order expires                                                                           |
+| lockScriptHash  | H160            | Lock script hash provided by maker, which should be written in every output with the order                |
+| parameters      | Bytes[]         | Parameters provided by maker, which should be written in every output with the order                      |
+
+To make a point-to-point order, put a zero on the `assetAmountFee` field.
+To write an order on a transfer transaction, the order should be wrapped once more, to `OrderOnTransfer`.
+
+## OrderOnTransfer format
+
+If there are inputs and outputs with the same order, it is wasteful to put the order in every input/output. Therefore, orders are wrapped into `OrderOnTransfer`.
+
+|       Name      |    Data Type    |                                                Description                                                |
+|-----------------|-----------------|-----------------------------------------------------------------------------------------------------------|
+| order           | Order           | The order to write on the transfer transaction                                                            |
+| spentAmount     | U64             | The spent amount of `assetTypeFrom` of the order in the transfer transaction                              |
+| inputIndices    | Index[]         | The indices of the transfer inputs which are protected by the order                                       |
+| outputIndices   | Index[]         | The indices of the transfer outputs which are protected by the order                                      |
+
+And the format of transfer transaction is as shown below.
+
+|      Name     |       Data Type       |
+|---------------|-----------------------|
+| networkId     | NetworkId             |
+| burns         | AssetTransferInput[]  |
+| inputs        | AssetTransferInput[]  |
+| outputs       | AssetTransferOutput[] |
+| orders        | OrderOnTransfer[]     |
+
+## How to support partial fills?
+
+As described before, with the 10-gold-to-100-silver order, it is possible to make a transfer transaction which has a 10-gold input that results in a 5-gold and 50-silver output, tagged with the same order. (Other inputs/outputs should be provided by a taker or a relayer). After this transaction, an asset contains the hash of the *consumed order*, which is 5-gold-to-50-silver order, not 10-gold-to-100-silver order. In order to use the 5-gold output, provide the 5-gold-to-50-silver order with the same information except for the `assetAmountFrom` and the `assetAmountTo` field. Neither a lock script nor a unlock script is needed to the 5-gold input. CodeChain will compare the order of an input and the order of a corresponding previous output, and will not run VM on the order if those orders are same.
```

Comment:
As described before, with the 10-gold-to-100-silver order, -> As described above in the 10-gold-to-100-silver order,

Reply:
Neither a lock script nor a unlock script is needed to the 5-gold input. -> Neither a lock script nor an unlock script is needed for the 5-gold input.

Reply:
CodeChain will compare the order of an input and the order of a corresponding previous output, and will not run VM on the order if those orders are same. -> CodeChain will compare the order of an input and the order of the corresponding previous output, and will not run VM on the order if those orders are identical.